python/pyspark/sql/connect/session.py

```diff
@@ -1041,11 +1045,17 @@ def _start_connect_server(master: str, opts: Dict[str, Any]) -> None:
             init_opts.update(opts)
             opts = init_opts
 
+            token = str(uuid.uuid4())
+
             # Configurations to be overwritten
             overwrite_conf = opts
             overwrite_conf["spark.master"] = master
             overwrite_conf["spark.local.connect"] = "1"
+            # When running a local server, always use an ephemeral port
+            overwrite_conf["spark.connect.grpc.binding.port"] = "0"
```

Comment:
I think we shouldn't mix up with the improvement and this auth change. Let's revert this change.

Reply:
Yeah I just had this in branch as I was messing with things, will revert

Reply:
Reverted